### ISSUE TITLE
Scrolling in browser slow/laggy

### ISSUE THREAD

**Issue by slfhstr** [kasmtech/workspaces-issues#112]
Trying out the Community edition before signing up
Installed Kasm on VPS with 32Gb ram
Have adjusted the image settings for Ubuntu Desktop to 4 cores and 8Gb RAM
Viewing a web page in Firefox in Ubuntu desktop seems slow to load and slow to scroll, very laggy.
`free -h` reports the image is not using the available allocated RAM
```
              total        used        free      shared  buff/cache   available
Mem:           31Gi       1.5Gi        27Gi       125Mi       2.8Gi        29Gi
Swap:         6.0Gi       2.0Mi       6.0Gi
```
Is there a performance tweak that can be implemented ?

**Comment by j-travis** [comment]
Hello,
how does your experience compare to loading a demo session from our site:
https://kasmweb.com/demo.html

**Comment by slfhstr** [comment]
The demo site certainly works better than my install.
Not as smooth as native browser but then I don't expect that.
My installed Kasm certainly not as performant as your demo site

**Comment by j-travis** [comment]
I see.
For reference:  our demo session run on Digital Ocean Premium Intel droplets. They are configured with 2cores and 2gb of ram. The demo service is also configured to give you a session in the region closest to you ( US East , US West, Germany, or India)

The main things that impact performance are
- Storage I/O. Ensure you are using SSD or better . Standard mechanical disks are significantly slower . For reference the DO Premium droplets use NVME. On slow storage you'll notice it takes a long time for sessions and programs to start.
- CPU core count/speed.  The more cores and the performance of the single core processor will improve the experience. You'll need to look at the configuration of your VPS . You'll also want to know if your CPU is shared with other tenants or if its dedicated. 
- Network Latency- You want as much bandwidth and minimal latency between you and your VPS.  The better the connection the smoother the experience when interacting with the session

**Comment by slfhstr** [comment]
Many thanks.
Wow, demo only has 2 core and 2Gb !  Impressive.
My VPS is on NVMe, but VPS is (like most VPS) on a shared platform.
Tried my Kasm installation using latest Safari and Firefox on client : both laggy.
Physically I (my client session) am in Ukraine, connecting to VPS in London.
But network latency is ok on normal web activity.
Maybe I will try Kasm on a different VPS.

**Comment by j-travis** [comment]
One other thing to mention: Or demo sessions use Session Staging which is a feature of workspaces :https://kasmweb.com/docs/latest/guide/staging.html

It works by pre-creating the session containers, and then assigns the already running container to the user when they request it.  The makes loading a session from the user perspective much faster as they don't have to wait for the container to be created. It also has a side benefit that the processes within the container have already started and initialized . For example when using a Chrome session, Chrome might normally take a few seconds to start and load the first page- that would be all be done beforehand in a staged session.

**Comment by slfhstr** [comment]
OK thank you, interesting.
Load speed for container is not bad.
I'm used to AWS Workspace which I would say is slower to load.
App opening is also acceptable, at least very understandable.
It's just the browser page navigation which is suffering.
Just fired up a VPN terminating in London where my Kasm VPS is located, and there is a small improvement (I appreciate VPNs affect speed anyway).
So maybe there is a latency issue.
I will do some more tests.
I will look into session staging later.

**Comment by slfhstr** [comment]
Update FYI : 
I installed Kasm on a different VPS
Chose Germany as location matching one of your locations
Although lower RAM and cores than my original test, it is a lot more performant.
So possibly --> probably the original VPS depends on more shared resources.
Session opening and app loading is much faster.
Browser page load much improved, page scroll definitely improved.
Maybe not quite as fast as your demo, but close.
Certainly within useable range.

Only problem I am getting is no sound in browser playing video. 
I understand that it is not supported on mobile browser, but I inferred desktop usage would be OK.
Is that a configuration setting somewhere ?  Or it is not supported anyway ?
